src/main/java/com/aws/greengrass/clientdevices/auth/PermissionEvaluationUtils.java

```diff
@@ -145,7 +147,19 @@ private static boolean compareResource(Resource requestResource, String policyRe
             return true;
         }
 
-        return ANY_REGEX.equals(policyResource);
+        if (ANY_REGEX.equals(policyResource)) {
+            return true;
+        }
+
+        // match wildcards
+        String rawPattern = Arrays.stream(policyResource.split("\\*"))
+                .map(Pattern::quote).collect(Collectors.joining(".*"));
+        if (policyResource.endsWith("*")) {
+            rawPattern += ".*";
+        }
+        Pattern pattern = Pattern.compile(rawPattern);
```

Comment:
we do not want to do this. Yes, this works, but it is treating the full input as regex which is a very bad practice. We need to parse the wildcards ourselves like in: https://github.com/aws-greengrass/aws-greengrass-nucleus/blob/b67b93ec970a9e77f143ecea6395abcf3c944e23/src/main/java/com/aws/greengrass/authorization/WildcardTrie.java#L25

Reply:
Especially since this code seems to run on every lookup, the trie would be a lot faster so long as you only build/update the trie once and not every lookup.

Reply:
Will refactor to trie

Reply:
Hey, are there any news or a timeline on this PR?

Reply:
@mschwab-bhs no updates on this, no. We hope to come back to these features soon. Thank you for your interest.